File: Segundo período/AED-2/TP01-parte1/Palindromo.c

/*
 *  Autor: Luiz Junio <luisjuniorbr@gmail.com>
 *  Enunciado:
 *   Palindromo em C - Na linguagem C, crie um metodo iterativo que recebe uma string como
 *   parametro e retorna true se essa e um palindromo. Na saida padrao, para cada linha de entrada,
 *   escreva uma linha de saida com SIM / NAO indicando se a linha e um palindromo.
 *   Observacao: Uma linha de entrada pode ter caracteres nao letras.
 */

#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int ehPalindromo(char palavra[])
{//Inicio ehPalindromo
    int palindromo = 1;//TRUE
    int letra = 0;
    int ultimo = strlen(palavra) - 1;//Posicao da ultima letra

    while(letra < strlen(palavra) / 2 && palindromo)
    {//Inicio while
        if(palavra[letra] != palavra[ultimo - letra]){
            palindromo = 0;//FALSE
        }
        letra++;
    }//Fim while
    return palindromo;
}//Fim ehPalindromo

int main()
{//Inicio main
    char palavra[1000][1000];
    int numEntrada = 0;
    do{//leitura da entrada padrao
        scanf(" %[^\n]s", palavra[numEntrada]);
        //Esse scanf irá ler a string, incluindo espaços.
        //O espaço antes de % serve para limpar o buffer do teclado
    }while(strcmp(palavra[numEntrada++], "FIM") != 0);//Repete enquanto não for igual a FIM

    numEntrada--; //Desconsiderar ultima linha contendo a palavra FIM

    for(int i = 0; i < numEntrada; i++)
    {//Inicio for
        if(ehPalindromo(palavra[i])){
            printf("SIM\n");
        }
        else{
            printf("NAO\n");
        }
    }//Fim for

}//Fim main 

File: Segundo período/AED-2/TP01-parte1/Palindromo.java
/*
    Autor: Luiz Junio <luisjuniorbr@gmail.com>
    Enunciado:
    Palindromo - Crie um metodo iterativo que recebe uma string como parametro e retorna true
    se essa e um palindromo. Na saida padrao, para cada linha de entrada, escreva uma linha de
    saida com SIM / NAO indicando se a linha e um palindromo.
*/


public class Palindromo
{//Inicio classe Palindromo     
    public static void main(String[] args)
    {//Inicio main
        String[] palavra = new String[1000];
        int numEntrada = 0;

        do{//Leitura da entrada padrao  
            palavra[numEntrada] = MyIO.readLine();
        }while (!ehIgual(palavra[numEntrada++],"FIM"));

        numEntrada--;   //Desconsiderar ultima linha contendo a palavra FIM

        //Impressao da saida
        for(int i = 0; i < numEntrada; i++)
        {//Inicio for
            if(ehPalindromo(palavra[i])){
                MyIO.println("SIM");
            }
            else MyIO.println("NAO");
        }//Fim for
    }//Fim main 

    public static boolean ehIgual(String p1, String p2)
    {//Inicio ehIgual
     /***************************************************************
      * Nome do método: ehIgual
      * Data da elaboração: 08/08/2018
      * Data da última alteração: 08/08/2018
      * Autor: Luiz Junio <luisjuniorbr@gmail.com>
      * Contexto de ação: Verifica se uma String é igual a outra.
      * Valor gerado: verdadeiro se forem iguais, falso caso contrário.
      ****************************************************************
      */ 
        boolean igual = true;
        int i = 0;
        if(p1.length() != p2.length()) igual = false;
        else while(i < p1.length() && igual){
                if(p1.charAt(i) != p2.charAt(i))//Comparacao caractere por caractere
                    igual = false;
                i++;
        }//Fim while
        return igual;

    }//Fim ehIgual

    public static boolean ehPalindromo(String palavra)
    {//Inicio ehPalindromo
     /***************************************************************
      * Nome do metodo: ehPalindromo
      * Data da elaboração: 08/08/2018
      * Data da ultima alteração: 08/08/2018
      * Autor: Luiz Junio <luisjuniorbr@gmail.com>
      * Contexto de ação: Verificar se uma String pode ser lida da esquerda
      * para a direita, e vice-versa.
      * Valor gerado: verdadeiro se for um paindromo, falso caso contrário.
      ****************************************************************
      */ 

        boolean palindromo = true;
        int letra = 0;
        int ultimo = palavra.length() - 1;//posicao da ultima letra

        while(letra < palavra.length() / 2 && palindromo){
            if(palavra.charAt(letra) != palavra.charAt(ultimo - letra)){ 
                palindromo = false;
            }//Fim if 
            letra++;
        }//Fim while
        return palindromo;
    }//Fim ehPalindromo

}//Fim classe Palindromo
